### PR TITLE
fix: Fix e2e after xui release

### PR DESCRIPTION
**Pull request by tomaszpowroznik**

### CODE REVIEW

File: e2e/pages/events/addStatementOfServiceEvent.page.js [hunk 1]
```diff
@@ -35,9 +30,8 @@ module.exports = {
     I.click(this.fields(elementIndex).recipients.addressCheck);
     this.enterRecipientsAddress(elementIndex, recipients);
     I.fillField(this.fields(elementIndex).recipients.documents, recipients.documents);
-    I.fillField(this.fields(elementIndex).recipients.date.day, recipients.date.day);
-    I.fillField(this.fields(elementIndex).recipients.date.month, recipients.date.month);
-    I.fillField(this.fields(elementIndex).recipients.date.year, recipients.date.year);
+    const dateSelector = `(//*[contains(@class, "collection-title")])[${ elementIndex + 1 }]/parent::div//*[contains(@class,"form-date")]`;
```
Comment (lewisbirks) [inline e2e/pages/events/addStatementOfServiceEvent.page.js:33]: could this replace the date entry in the fields function?

File: e2e/pages/events/addGatekeepingOrderEvent.page.js [hunk 2]
```diff
@@ -60,7 +60,8 @@ module.exports = {
     I.fillField(this.fields.customDirection.fields(elementIndex).description, direction.description);
     I.selectOption(this.fields.customDirection.fields(elementIndex).assignee, direction.assignee);
     I.click(this.fields.customDirection.fields(elementIndex).dueDateType);
-    await I.fillDate(direction.dueDate, this.fields.customDirection.fields(elementIndex).date);
+    const dateSelector = `(//*[contains(@class, "collection-title")])[${ elementIndex + 1 }]/parent::div//*[@id="dateToBeCompletedBy"]`;
```
Comment (lewisbirks) [inline e2e/pages/events/addGatekeepingOrderEvent.page.js:63]: if the date is no longer being used in the custom directions object, can this replace it?

File: e2e/pages/events/closeTheCaseEvent.page.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const {I} = inject();
 
 const fields = {
-  date: '#closeCase_date',
+  date: '#date',
```
Comment (lewisbirks) [inline e2e/pages/events/closeTheCaseEvent.page.js:4]: not sure I understand this one either

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java [hunk 4]
```diff
@@ -249,7 +249,7 @@ public HearingBooking getHearingBooking(UUID id, List<Element<HearingBooking>> h
 
     public void findAndSetPreviousVenueId(CaseData caseData) {
         if (isNotEmpty(caseData.getHearingDetails()) && caseData.getPreviousHearingVenue() != null
-            && caseData.getPreviousHearingVenue().getUsePreviousVenue().equals(YES.getValue())) {
+            && YES.getValue().equals(caseData.getPreviousHearingVenue().getUsePreviousVenue())) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageHearingsService.java:252]: would it be worth adding a unit test to cover the bug?

File: e2e/fragments/directions.js [hunk 5]
```diff
@@ -6,13 +6,13 @@ module.exports = {
...
         title: `#${party}_${index}_directionType`,
         description: `#${party}_${index}_directionText`,
-        dueDate: `#${party}_${index}_dateToBeCompletedBy`,
       },
     };
   },
 
   async enterDate(party, dueDate, index = 0) {
-    await I.fillDateAndTime(dueDate, this.fields(party, index).direction.dueDate);
+    const dateSelector = `(//*[contains(@class, "collection-title")])[${ index + 1 }]/parent::div//*[contains(@class,"form-date")]`;
```
Comment (Rob-Doherty) [inline e2e/fragments/directions.js:14]: is it worth overwriting the `this.fields(party, index).direction.dueDate ` value with this new xpath? to follow the existing pattern?

File: e2e/pages/events/addGatekeepingOrderEvent.page.js (hunk 2 above)
Comment (Rob-Doherty) [inline e2e/pages/events/addGatekeepingOrderEvent.page.js:63]: again, is it worth overwriting the `this.fields.customDirection.fields(elementIndex).date` value with this new xpath? to follow the existing pattern?

File: e2e/pages/events/addStatementOfServiceEvent.page.js (hunk 1 above)
Comment (Rob-Doherty) [inline e2e/pages/events/addStatementOfServiceEvent.page.js:33]: again, is it worth overwriting the `this.fields(elementIndex).recipients.date ` value with this new xpath? to follow the existing pattern?

File: e2e/pages/events/enterOthersEvent.page.js [hunk 6]
```diff
@@ -23,22 +18,20 @@ module.exports = {
...
-    const elementIndex = await this.getActiveElementIndex();
+    const elementIndex = await this.getActiveElementSelector();
 
     await I.runAccessibilityTest();
     I.fillField(this.fields(elementIndex).name, other.name);
-    I.click(this.fields(elementIndex).DOB.day);
-    I.fillField(this.fields(elementIndex).DOB.day, other.DOB.day);
-    I.fillField(this.fields(elementIndex).DOB.month, other.DOB.month);
-    I.fillField(this.fields(elementIndex).DOB.year, other.DOB.year);
+    const dateSelector = `(//*[@id="DOB"])[${await this.getActiveElementIndex()}]`;
```
Comment (Rob-Doherty) [inline e2e/pages/events/enterOthersEvent.page.js:33]: again, is it worth overwriting the `this.fields(elementIndex).DOB` value with this new xpath? to follow the existing pattern?

**Review by Rob-Doherty — COMMENTED** [review]
just some suggestions around page object pattern consistency. the changes themselves look good

**Review by lewisbirks — DISMISSED** [review]

**Review by Rob-Doherty — APPROVED** [review]
looks good to me 👍

**Pull request merged**